testsuite/tests/ui/test_unique_invoice_id.py

```diff
@@ -0,0 +1,65 @@
+"""
+rewrite of /spec/ui_specs/billing/unique_invoice_id_spec.rb
+For testing this issue, we don't need to use UI
+"""
+from datetime import date, timedelta
+import pytest
+from threescale_api.resources import InvoiceState, Account, ApplicationPlan
+
+pytestmark = pytest.mark.issue("https://issues.jboss.org/browse/THREESCALE-1203")
+
+NACCOUNTS = 10
+NEXT_DAY = (date.today() + timedelta(days=1)).isoformat()
+
+
+@pytest.fixture(scope="module")
+def paid_app_plan(service, custom_app_plan) -> ApplicationPlan:
+    """Application plan on default service with some setup fee"""
+    return custom_app_plan(dict(name="paidtestplan", setup_fee="100"), service=service)
+
+
+@pytest.fixture(scope="module")
+def custom_paid_account(paid_app_plan, custom_account, custom_application):
+    """Returns function for custom account creation"""
+    def _custom_paid_account(num: int) -> Account:
+        """
+        Makes number of accounts with applications that use paid app plan.
+        :param num: Number of accounts to create
+        """
+        acc_params = dict(name=f"test-{num}", username=f"test-{num}",
+                          email=f"{num}@anything.invalid", password="123456")
+        account = custom_account(acc_params)
+        custom_application(dict(name=f"app-{num}", description="desc",
+                                plan_id=paid_app_plan.entity_id), custom_account=account)
+        return account
+    return _custom_paid_account
+
+
+@pytest.fixture(scope="function")
+def create_invoice(custom_paid_account, master_threescale, threescale,
+                   provider_account, request):
+    """Creates NACCOUNTS accounts and for each account charge invoice with master_api"""
+    for i in range(NACCOUNTS):
+        account = custom_paid_account(i)
+        master_threescale.tenants.trigger_billing_account(provider_account, account, NEXT_DAY)
+
+    def clean_invoice():
+        for i in range(NACCOUNTS):
+            acc: Account = threescale.accounts.read_by_name(f"test-{i}")
+            for invoice in threescale.invoices.list_by_account(acc):
+                invoice.state_update(InvoiceState.CANCELLED)
+
+    request.addfinalizer(clean_invoice)
```

Comment:
It would be best if you added the finalizer before creating the accounts

Reply:
fixed